src/Ledger/Ledger/Properties.agda

```diff
@@ -82,29 +87,536 @@ instance
+    (tx : Tx) (let open Tx tx renaming (body to txb)) (let open TxBody txb)
+    (Γ : LEnv) (let open LEnv Γ renaming (pparams to pp))
+    (s : LState)
+    where
+
+    f : GovActionID × GovActionState → DepositPurpose
+    f = λ (id , _) → GovActionDeposit id
+
+    open PParams pp using (govActionDeposit)
+
+    utxoSt : UTxOState
+    utxoSt = LState.utxoSt s
+
+    govSt : GovState
+    govSt = LState.govSt s
+
+    utxoDeps : DepositPurpose ⇀ Coin
+    utxoDeps = getDeps s
+
+    depUpdate_withIsValid≡_ : LState → Bool → DepositPurpose ⇀ Coin
+    depUpdate s withIsValid≡ true = updateDeposits pp txb (getDeps s)
+    depUpdate s withIsValid≡ false = getDeps s
+
+    STS→utxoDeps≡' : ∀ {s' : LState} → Γ ⊢ s ⇀⦇ tx ,LEDGER⦈ s'
+                    → getDeps s' ≡ depUpdate s withIsValid≡ isValid
+
+    STS→utxoDeps≡' {.(⟦ utxoSt' , _ , _ ⟧ˡ)} (LEDGER-V {utxoSt' = utxoSt'}
+      (val , UTXOW-inductive⋯ _ _ _ _ _ _ _ (UTXO-inductive⋯ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ UTXOS-sts) , _ , _)) =
+        trans (⊢utxo-valid UTXOS-sts) (sym (cong (depUpdate s withIsValid≡_) val))
+          where
+          ⊢utxo-valid : record { slot = slot ; pparams = pp } ⊢ LState.utxoSt s ⇀⦇ tx ,UTXOS⦈ utxoSt'
+                        → UTxOState.deposits utxoSt' ≡ updateDeposits pp txb (getDeps s)
+          ⊢utxo-valid (_⊢_⇀⦇_,UTXOS⦈_.Scripts-Yes _) = refl
+          ⊢utxo-valid (_⊢_⇀⦇_,UTXOS⦈_.Scripts-No u) = contradiction (trans (sym val) (proj₂ u)) (λ ())
+
+    STS→utxoDeps≡' {.(⟦ utxoSt' , _ , _ ⟧ˡ)} (LEDGER-I {utxoSt' = utxoSt'}
+      (¬val , UTXOW-inductive⋯ _ _ _ _ _ _ _ (UTXO-inductive⋯ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ UTXOS-sts))) =
+        trans (⊢utxo-invalid UTXOS-sts) (sym (cong (depUpdate s withIsValid≡_) ¬val))
+          where
+          ⊢utxo-invalid : record { slot = slot ; pparams = pp } ⊢ LState.utxoSt s ⇀⦇ tx ,UTXOS⦈ utxoSt'
+                        → UTxOState.deposits utxoSt' ≡ getDeps s
+          ⊢utxo-invalid (_⊢_⇀⦇_,UTXOS⦈_.Scripts-Yes u) = contradiction (trans (sym ¬val) (proj₂ u)) (λ ())
+          ⊢utxo-invalid (_⊢_⇀⦇_,UTXOS⦈_.Scripts-No _) = refl
+
+    STS→utxoDeps≡ᵉ' : ∀ {s' : LState} → Γ ⊢ s ⇀⦇ tx ,LEDGER⦈ s'
+                    → dom (getDeps s') ≡ᵉ dom (depUpdate s withIsValid≡ isValid)
+    STS→utxoDeps≡ᵉ' utxos-sts = let open IsEquivalence (≡ᵉ-isEquivalence{DepositPurpose}) in
+      reflexive (cong dom (STS→utxoDeps≡' utxos-sts))
+
+    STS→utxoDeps≡ : ∀ {s' : LState} → Γ ⊢ s ⇀⦇ tx ,LEDGER⦈ s'
+                    → isValid ≡ true → getDeps s' ≡ updateDeposits pp txb (getDeps s)
+    STS→utxoDeps≡ utxo-sts txvalid = trans (STS→utxoDeps≡' utxo-sts) (cong (depUpdate s withIsValid≡_) txvalid)
+
+    STS→utxoDeps≡ᵉ : ∀ {s' : LState} → Γ ⊢ s ⇀⦇ tx ,LEDGER⦈ s'
+                    → isValid ≡ true → dom (getDeps s') ≡ᵉ dom (updateDeposits pp txb (getDeps s))
+    STS→utxoDeps≡ᵉ utxo-sts txvalid = let open IsEquivalence (≡ᵉ-isEquivalence{DepositPurpose}) in
+      reflexive (cong dom (STS→utxoDeps≡ utxo-sts txvalid))
```

Comment:
This is unused

Reply:
No, it's used in the first line of the proof of the main theorem, at or around line 566:

```agda
      begin
      filterˢ isGADeposit (dom utxoDeps')
        ≈⟨ filter-pres-≡ᵉ (STS→utxoDeps≡ᵉ utxosts tx-valid) ⟩
        ...
```

Reply:
However, I have since in-lined the proof of `STS→utxoDeps≡ᵉ` in the one place where it was used... so I will now delete `STS→utxoDeps≡ᵉ`.